eunji/src/components/Select.tsx

```diff
@@ -0,0 +1,311 @@
+import { useRouter } from "next/router";
+import React, { ChangeEvent, useEffect, useState } from "react";
+import styled from "styled-components";
+
+// 좌석 선택을 위한 배열 생성
+const selectArray: Array<string> = new Array<string>();
+
+export default function Select() {
+    // 사람의 수
+    const [person, setPerson] = useState<number>(0);
+    // 영화의 가격
+    const [price, setPrice] = useState<number>(600);
+    // 총 가격
+    const [total, setTotal] = useState<number>(3);
+    // 영화 이름
+    const [name, setName] = useState<string>('Aladin');
+    // 영화 이미지
+    const [poster, setPoster] = useState<string>('');
+    // router query를 위하여 정의
+    const router = useRouter();
+
+    // 영화 선택될 때마다 가격, 영화제목 바뀌게 하기
+    const onSelectChange = (e: ChangeEvent<HTMLSelectElement>) => {
+        setPrice(Number(e.target.value));
+
+        // 만약 value가 600, 700, 800이면 setName을 그 값에 해당하는 영화 이름으로 바꿔라.
+        if (e.target.value == '600') {
+            setName(String('Aladin'));
+            setPoster("movie1.jpg");
+        }
+        if (e.target.value == '700') {
+            setName(String('Beauty and the Beast'));
+        }
+        if (e.target.value == '800') {
+            setName(String('범죄도시2'));
+        }
+    };
+
+    // 선택한 인원만큼 가격 증가
+    useEffect(() => {
+        const realPerson = person;
+        setTotal(price * realPerson);
+    }, [price, person]);
+
+    // 좌석 선택 가능하게 하기
+    const SelectChange = (e: React.MouseEvent<HTMLButtonElement>) => {
+        const selectStyle = e.currentTarget;
+        const changeSeat = e.currentTarget.innerHTML;
+
+        // 만약 changeSeat이 selectArray에 포함된다면
+        if (selectArray.includes(changeSeat)) {
+            const indexSeat = selectArray.indexOf(changeSeat);
+            // indexSeat를 삭제하고(기본값)
+            delete selectArray[indexSeat];
+            // 배경색을 변경하고
+            selectStyle.style.backgroundColor = "#666";
+            // 값을 줄여라
+            setPerson(person - 1);
+        }
+        // 그렇지 않으면
+        else {
+            // selectArray의 changeSeat을 클릭했을 때
+            selectArray.push(changeSeat);
+            // 배경색을 변경하고
+            selectStyle.style.backgroundColor = "#ad1712";
+            // 값을 늘려라
+            setPerson(person + 1);
+        }
+    }
+
+    // 혹시 -값이나 좌석을 선택하지 않았을 시 0으로 리셋
+    const onReset = () => {
+        setPerson(0);
+        setPrice(0);
+    }
+
+    // 좌석 선택 후 결제창으로 넘어가기
+    const onClickPurchase = () => {
+        // 만약 선택을 안 하면
+        if (person <= 0) {
+            // 해당 알림을 띄우고
+            alert('좌석을 선택해 주세요.');
+            // 0으로 리셋해 주고
+            onReset();
+            // 알라딘의 기본값인 600으로 변경해라
+            setPrice(600);
+            return;
+        }
+        // 그렇지 않으면(선택을 하면)
+        else router.push({
+            // purchase로 이동하고
+            pathname: '/purchase',
+            // 해당 값들을 purchase로 넘겨라
+            query: { poster: poster, name: name, person: person, price: total },
+        });
```

Comment:
위에서 return을 하고 있기 때문에 여기서 else가 하는 일은 사실 없을 거 같아요
```suggestion
            return;
        }
        // 그렇지 않으면(선택을 하면)
       router.push({
            // purchase로 이동하고
            pathname: '/purchase',
            // 해당 값들을 purchase로 넘겨라
            query: { poster: poster, name: name, person: person, price: total },
        });
```

이렇게 작성하는게 `early return`이라는 클린 코드 기법 중 하난데, 취향을 타긴 하지만 저는 좋아합니다

Reply:
else를 빼면 안 되는 줄 알았어요ㅠ 덕분에 새로운 기법 알아갑니다!